### PR TITLE
perf(trie-router): optimize and remove unnecessary processes

### PR DESCRIPTION
**Pull request by EdamAme-x**
1. The following process is not necessary and has been removed.

https://github.com/honojs/hono/pull/3647/files#diff-bf1549ba9405f87c376d2d338883f54bfeb6a91f40b28e62dd2ce5e9d4edd4e5L66-L69

2. Since it will be minified, I changed the name to be easy to understand.

Related: https://github.com/honojs/hono/pull/3596

https://github.com/honojs/hono/pull/3647/files#diff-bf1549ba9405f87c376d2d338883f54bfeb6a91f40b28e62dd2ce5e9d4edd4e5L85

3. Since only the type number is used for the key, an empty object should be created instead of `Object.create(null)`. It is faster and smaller.

Node and Bun (8x~ faster)
```llvm
runtime: node 20.18.0 (x64-linux)

benchmark              avg (min … max) p75   p99    (min … top 1%)
-------------------------------------- -------------------------------
= Object.create(null)    16.67 ns/iter  15.13 ns  █                   
                 (13.54 ns … 96.35 ns)  42.71 ns ▇█▂▁▂▂▁▁▂▁▁▁▁▁▁▁▁▁▁▁▁
= {}                      2.41 ns/iter   2.22 ns  █                   
                  (1.84 ns … 95.40 ns)   8.21 ns ▁█▂▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
```

```llvm
runtime: deno 2.0.5 (x86_64-unknown-linux-gnu)

benchmark              avg (min … max) p75   p99    (min … top 1%)
-------------------------------------- -------------------------------
= Object.create(null)    16.64 ns/iter  14.93 ns  █                   
                (12.03 ns … 137.93 ns)  33.21 ns ▁██▃▁▁▁▁▁▁▁▁▁▁▅▂▁▁▁▁▁
= {}                      2.70 ns/iter   2.19 ns █                    
                  (2.03 ns … 62.84 ns)  16.17 ns █▂▂▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
```
In Bun, the speed of both is almost the same.

https://github.com/honojs/hono/pull/3647/files#diff-bf1549ba9405f87c376d2d338883f54bfeb6a91f40b28e62dd2ce5e9d4edd4e5L95

~~4. Multiple index accesses are eliminated and the size is reduced.~~

~~https://github.com/honojs/hono/pull/3647/files#diff-bf1549ba9405f87c376d2d338883f54bfeb6a91f40b28e62dd2ce5e9d4edd4e5R130~~

5. Remove as it is not a necessary check

https://github.com/honojs/hono/pull/3647/files#diff-bf1549ba9405f87c376d2d338883f54bfeb6a91f40b28e62dd2ce5e9d4edd4e5L178

### The author should do the following, if applicable

- [ ] Add tests
- [x] Run tests
- [x] `bun run format:fix && bun run lint:fix` to format the code
- [ ] Add [TSDoc](https://tsdoc.org/)/[JSDoc](https://jsdoc.app/about-getting-started) to document the code

### CODE REVIEW

**Comment by yusukebe** [comment]
Hi @EdamAme-x !

It looks awesome! I'd like to know how much this change affects performance. Did you benchmark it? And how much did you reduce the bundle size?

PS.

To benchmark the req/res performance, you can follow the method used in this repo: https://github.com/SaltyAom/bun-http-framework-benchmarkhttps://github.com/SaltyAom/bun-http-framework-benchmark

**Comment by EdamAme-x** [comment]
# Report

Revert (4): The bundle size is smaller, but we got strange benchmark results that one index access is slower than multiple index accesses, so we reverted.
```llvm
runtime: deno 2.0.5 (x86_64-pc-windows-msvc)

benchmark              avg (min … max) p75   p99    (min … top 1%)
-------------------------------------- -------------------------------
Once                    193.31 ps/iter 195.31 ps ▇  █
                (170.90 ps … 70.87 ns) 341.80 ps █▁▁█▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
Multi                   136.00 ps/iter 146.48 ps     █
                 (97.66 ps … 22.39 ns) 219.73 ps ▁▁▁▁█▁▁▁▄▁▁▁▂▁▁▁▂▁▁▁▁
```

## Bundle Size

19.34kb => 19.28kb (1kb = 1024b)
```llvm
gitpod@honojs-hono-0oouxesc90f:/workspace/hono/sandbox$ ls compare -l
total 40
-rw-r--r-- 1 gitpod gitpod 19288 Nov  9 07:09 after.js
-rw-r--r-- 1 gitpod gitpod 19344 Nov  9 07:11 before.js
```

## Benchmark

### In Deno
There was not much change.
(+0.03%)
```llvm
|  Framework       | Runtime | Average | Ping       | Query      | Body       |
| ---------------- | ------- | ------- | ---------- | ---------- | ---------- |
| after | deno | 41,225.407 | 50,386.78 | 38,903.5 | 34,385.94 |
| before | deno | 41,079.073 | 50,923.38 | 39,116.56 | 33,197.28 |
```

**Comment by yusukebe** [comment]
@EdamAme-x 

Have you ever tried to benchmark frameworks on Bun, not only Deno?

**Comment by EdamAme-x** [comment]
I'll benchmark with several approaches.

**Comment by EdamAme-x** [comment]
I measured the performance of router itself using node, bun, and mitata, but did not see any significant performance improvement that we could see.

File: src/router/trie-router/node.ts [hunk 1]
```diff
@@ -151,8 +154,10 @@ export class Node<T> {
           // '/hello/*/foo' => match /hello/bar/foo
           if (pattern === '*') {
             const astNode = node.children['*']
-            if (astNode) {
-              handlerSets.push(...this.#gHSets(astNode, method, node.params, Object.create(null)))
+            if (node.children['*']) {
```
Comment (yusukebe) [inline src/router/trie-router/node.ts:157]: Is this intended? `node.children['*']` can be `astNode`.

Comment (EdamAme-x) [reply src/router/trie-router/node.ts:157]: Sorry, I made a mistake in changing it.

**Comment by yusukebe** [comment]
@EdamAme-x 

Thanks. Anyway, this is a good improvement. I'll merge this now.

**Pull request merged**